.github/workflows/typechecks.yml

```diff
@@ -0,0 +1,43 @@
+name: Tests, Linter & Typecheck
+
+on: pull_request
+
+jobs:  
+  lint:
+    name: 'Linting Markdown'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+
+      - name: Use Node.js 16.7
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.7
+
+      - run: npm install -g markdownlint-cli@0.23.2
+      - run: markdownlint '**/*.md' --ignore node_modules
```

Comment:
add `markdownlint` into `devDependencies` in `package.json` and add an npm script that calls `markdownlint '**/*.md' --ignore node_modules`, and reference that script name here.

Reply:
The markdownlint-cli being installed as a `devDependencies` and called from the script was failing (locally it worked) inside of the GH action it was unable to find it. I checked  markdownlint-cli and markdownlint docs/README the examples looked liked this current setup for CI and was globally installed for local use. 
I alternatively got it to work inside the NPM script with NPX which I can also just put directly in the GH Action.
It does mean that it doesn't get cached with the other dependencies.

Reply:
We should figure out why. Not a fan of using `npx` for this.

Reply:
An alternate solution is to just not use this, and use `prettier --check` instead, I assume it does the same thing?

Reply:
> We should figure out why. Not a fan of using `npx` for this.

Just curious why? 

> An alternate solution is to just not use this, and use `prettier --check` instead, I assume it does the same thing?

I can look into this, I think your likely correct Prettier probably can handle this.

Reply:
> Just curious why not a fan of npx for this?
Because it doesn't solve the actual problem? We miss out on versioning, it doesn't get cached, etc.

Reply:
I tried Prettier out, it seems to work fine locally but it does make a lot of initial changes (especially CHANGELOG.md) mainly due to Prettier's default line length (80 chars) and adding some spaces. It does make the code examples in Markdown all uniform which a bit of a bonus.

Reply:
re: line length https://prettier.io/docs/en/options.html#prose-wrap

Reply:
> re: line length https://prettier.io/docs/en/options.html#prose-wrap

I can turn it off here: https://github.com/cloudflare/workers-types/blob/master/.prettierrc#L3 or pass it into the CLI should minimize the initial Prettier impact and remove the markdownlinter dep